client/lib/signup/actions.js

```diff
@@ -53,7 +53,15 @@ const SignupActions = {
 	},
 
 	processedSignupStep( step, errors, providedDependencies ) {
-		analytics.tracks.recordEvent( 'calypso_signup_actions_complete_step', { step: step.stepName } );
+		const { stepName } = step;
+
+		analytics.tracks.recordEvent( 'calypso_signup_actions_complete_step', { step: stepName } );
+
+		if ( stepName === 'survey-user' ) {
```

Comment:
We should avoid hardcoding step names wherever possible in signup. The steps can change at any moment and this code would stop working. 

For example there are currently two steps that are responsible for creating users - `user` and `survey-user`. The `user` step is used in some of the developer-only flows and will not fire the event.

I think the best way to do that is to add the code directly in [`step-actions.js`/`createAccount`](https://github.com/Automattic/wp-calypso/blob/1ae3209cb1169686f065087aa6a800afdc4bf9c0/client/lib/signup/step-actions.js#L181) where the code that handles the user creation request sits.